? comphelper/unxlngi6.pro.nodebug
? comphelper/inc/comphelper/unitbootstrap.hxx
? comphelper/source/unit
Index: comphelper/prj/build.lst
===================================================================
RCS file: /cvs/util/comphelper/prj/build.lst,v
retrieving revision 1.11
diff -u -p -u -r1.11 build.lst
--- comphelper/prj/build.lst	30 Oct 2006 09:31:48 -0000	1.11
+++ comphelper/prj/build.lst	20 Feb 2008 16:54:49 -0000
@@ -8,6 +8,7 @@ ph	comphelper\source\processfactory		nma
 ph	comphelper\source\property				nmake	-	all	ph_property ph_inc NULL
 ph	comphelper\source\streaming				nmake	-	all	ph_streaming ph_inc NULL
 ph	comphelper\source\compare				nmake	-	all	ph_compare ph_inc NULL
+ph	comphelper\source\unit				nmake	-	all	ph_unit ph_inc NULL
 ph  comphelper\source\officeinstdir         nmake   -   all ph_officeinstdir ph_inc NULL
 ph  comphelper\source\xml                   nmake   -   all ph_xml NULL
-ph  comphelper\util                         nmake   -   all ph_util ph_container ph_evtatmgr ph_misc ph_procfact ph_property ph_streaming ph_compare ph_officeinstdir ph_xml NULL
+ph  comphelper\util                         nmake   -   all ph_util ph_container ph_evtatmgr ph_misc ph_procfact ph_property ph_streaming ph_compare ph_officeinstdir ph_xml ph_unit NULL
Index: comphelper/prj/d.lst
===================================================================
RCS file: /cvs/util/comphelper/prj/d.lst,v
retrieving revision 1.13
diff -u -p -u -r1.13 d.lst
--- comphelper/prj/d.lst	24 Oct 2006 13:42:28 -0000	1.13
+++ comphelper/prj/d.lst	20 Feb 2008 16:54:49 -0000
@@ -12,3 +12,10 @@ mkdir: %_DEST%\inc%_EXT%\comphelper
 mkdir: %_DEST%\inc%_EXT%\cppuhelper
 ..\inc\comphelper\extract.hxx %_DEST%\inc%_EXT%\cppuhelper\extract.hxx
 ..\version.mk  %_DEST%\inc%_EXT%\comphelper\version.mk
+
+mkdir: %_DEST%\xml%_EXT%\unit-skel
+mkdir: %_DEST%\xml%_EXT%\unit-skel\data
+mkdir: %_DEST%\xml%_EXT%\unit-skel\data\org
+mkdir: %_DEST%\xml%_EXT%\unit-skel\data\org\openoffice
+..\source\unit\configmgrrc %_DEST%\xml%_EXT%\unit-skel\configmgrrc
+..\source\unit\Setup.xcu   %_DEST%\xml%_EXT%\unit-skel\data\org\openoffice\Setup.xcu
Index: comphelper/util/makefile.mk
===================================================================
RCS file: /cvs/util/comphelper/util/makefile.mk,v
retrieving revision 1.13
diff -u -p -u -r1.13 makefile.mk
--- comphelper/util/makefile.mk	20 Sep 2007 16:40:40 -0000	1.13
+++ comphelper/util/makefile.mk	20 Feb 2008 16:54:49 -0000
@@ -74,6 +74,23 @@ SHL1DEF=	$(MISC)$/$(SHL1TARGET).def
 DEF1NAME=	$(SHL1TARGET)
 DEFLIB1NAME=$(TARGET)
 
+# unit test helper library
+
+SHL2TARGET=unitbootstrap
+SHL2STDLIBS= \
+	$(SALLIB) \
+	$(SALHELPERLIB) \
+	$(CPPUHELPERLIB) \
+	$(CPPULIB) \
+	$(COMPHELPERLIB)
+SHL2DEPN=
+SHL2IMPLIB=	i$(SHL2_TARGET)
+SHL2USE_EXPORTS=ordinal
+SHL2LIBS=	$(SLB)$/unitbootstrap.lib
+# SHL1DEF=	$(MISC)$/$(SHL1TARGET).def
+# DEF1NAME=	$(SHL1TARGET)
+# DEFLIB1NAME=$(TARGET)
+
 # --- Targets ----------------------------------
 
 .INCLUDE : target.mk
--- /dev/null	2007-09-21 22:50:58.000000000 +0100
+++ comphelper/inc/comphelper/unitbootstrap.hxx	2008-02-20 16:37:16.000000000 +0000
@@ -0,0 +1,18 @@
+#ifndef UNIT_BOOTSTRAP_HXX
+#define UNIT_BOOTSTRAP_HXX
+
+#include "com/sun/star/lang/XMultiServiceFactory.hpp"
+#include "com/sun/star/uno/XComponentContext.hpp"
+
+class UnitBootstrap
+{
+    com::sun::star::uno::Reference< com::sun::star::uno::XComponentContext > mxContext;
+    com::sun::star::uno::Reference< com::sun::star::lang::XMultiServiceFactory > mxMSF;
+  public:
+    UnitBootstrap(const char *pSetup);
+    ~UnitBootstrap();
+    com::sun::star::uno::Reference< com::sun::star::lang::XMultiServiceFactory >
+        getMSF() { return mxMSF; }
+};
+
+#endif // UNIT_BOOTSTRAP_HXX
--- /dev/null	2007-09-21 22:50:58.000000000 +0100
+++ comphelper/source/unit/unitbootstrap.cxx	2008-02-20 16:59:30.000000000 +0000
@@ -0,0 +1,121 @@
+// To debug me use:
+// $ export ENVCFGFLAGS='-me -ti -tw -tp -td'
+
+#include "sal/config.h"
+#include <cstdlib>
+#include "com/sun/star/beans/XPropertySet.hpp"
+#include "com/sun/star/lang/XComponent.hpp"
+#include "com/sun/star/lang/XMultiComponentFactory.hpp"
+#include "com/sun/star/lang/XMultiServiceFactory.hpp"
+#include "com/sun/star/uno/Any.hxx"
+#include "com/sun/star/uno/Reference.hxx"
+#include "com/sun/star/uno/XComponentContext.hpp"
+#include "cppuhelper/component_context.hxx"
+#include "cppuhelper/servicefactory.hxx"
+#include "comphelper/processfactory.hxx"
+#include "osl/file.hxx"
+#include "osl/thread.h"
+#include "osl/process.h"
+#include "rtl/string.h"
+#include "rtl/bootstrap.h"
+#include "rtl/ustring.h"
+#include "rtl/ustring.hxx"
+#include "sal/types.h"
+#include "osl/file.hxx"
+#include "comphelper/unitbootstrap.hxx"
+
+namespace css = com::sun::star;
+
+static void
+setBootstrap (const char *pKey, rtl::OUString aValue)
+{
+    rtl::OUString aKey = rtl::OUString::createFromAscii( pKey );
+    rtl_bootstrap_set( aKey.pData, aValue.pData );
+}
+
+static rtl::OUString pathToURL(rtl::OUString aPath)
+{
+    rtl::OUString aURL;
+    if (osl::FileBase::getFileURLFromSystemPath(aPath, aURL) != osl::FileBase::E_None)
+        std::abort();
+    return aURL;
+}
+
+// Do all those exciting things to make the tests actually work ...
+UnitBootstrap::UnitBootstrap(const char *pSetup)
+{
+    // parse params ...
+    char const * f = pSetup;
+    rtl::OUString args (f, rtl_str_getLength(f), osl_getThreadTextEncoding());
+    sal_Int32 i = args.indexOf('#');
+    if (i < 0)
+        std::abort();
+    rtl::OUString aRDB(args.copy(0, i));
+    rtl::OUString aUserDir(args.copy(i + 1));
+
+    fprintf (stderr, "RDB: '%s'\n",
+             rtl::OUStringToOString (aRDB, RTL_TEXTENCODING_UTF8).getStr());
+    fprintf (stderr, "UserDir: '%s'\n",
+             rtl::OUStringToOString (aUserDir, RTL_TEXTENCODING_UTF8).getStr());
+
+    rtl_uString *curWd = NULL;
+    osl_getProcessWorkingDir( &curWd );
+
+    rtl::OUString aBase (curWd);
+
+    // Unless you do this, obviously you don't want your paths to be correct 
+    // or file access to work properly
+    setBootstrap ("BaseInstallation", aBase);
+    setBootstrap ("CFG_INIFILE", pathToURL (aUserDir) + rtl::OUString::createFromAscii( "/configmgrrc" ) );
+    setBootstrap ("UserInstallation", pathToURL (aUserDir));
+
+    rtl::OUString aSolver = (rtl::OUString::createFromAscii(getenv("SOLARVER")) +
+                             rtl::OUString::createFromAscii("/") +
+                             rtl::OUString::createFromAscii(getenv("INPATH")));
+    // Used in configmgrrc to locate the xml config
+    setBootstrap ("SOLVER", pathToURL (aSolver));
+
+    // Force vcl to be headless on Unix
+    putenv ((char *)(new rtl::OString("SAL_USE_VCLPLUGIN=svp"))->getStr());
+    putenv ((char *)(new rtl::OString("OOO_FORCE_DESKTOP=none"))->getStr());
+
+    css::uno::Reference< css::beans::XPropertySet > xFactory(
+        cppu::createRegistryServiceFactory(aRDB), css::uno::UNO_QUERY_THROW);
+    css::uno::Reference< css::uno::XComponentContext > xContext(
+        xFactory->getPropertyValue(
+            rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("DefaultContext"))),
+        css::uno::UNO_QUERY_THROW);
+
+    rtl::OUString regurl = rtl::OUString();
+    cppu::ContextEntry_Init entry(
+        rtl::OUString(
+            RTL_CONSTASCII_USTRINGPARAM(
+                "/modules/com.sun.star.configuration/bootstrap/Strata")),
+        css::uno::makeAny(
+            rtl::OUString(
+                RTL_CONSTASCII_USTRINGPARAM(
+                    "com.sun.star.comp.configuration.backend.LocalStratum:"))
+            + regurl)); 
+    mxContext = cppu::createComponentContext(&entry, 1, xContext);
+    if (!mxContext.is())
+        std::abort();
+
+    css::uno::Reference< css::lang::XMultiComponentFactory > xMCF (
+        mxContext->getServiceManager(), css::uno::UNO_QUERY_THROW);
+    xMCF->createInstanceWithContext(
+        rtl::OUString::createFromAscii(
+            "com.sun.star.configuration.ConfigurationProvider"),
+        mxContext);
+
+    mxMSF = css::uno::Reference< css::lang::XMultiServiceFactory >(
+        mxContext->getServiceManager(), css::uno::UNO_QUERY );
+    comphelper::setProcessServiceFactory( mxMSF );
+}
+
+UnitBootstrap::~UnitBootstrap()
+{
+    css::uno::Reference< css::lang::XComponent >(
+        mxMSF, css::uno::UNO_QUERY_THROW)->dispose();
+    css::uno::Reference< css::lang::XComponent >(
+        mxContext, css::uno::UNO_QUERY_THROW)->dispose();
+}
--- /dev/null	2007-09-21 22:50:58.000000000 +0100
+++ comphelper/source/unit/makefile.mk	2008-02-20 16:49:55.000000000 +0000
@@ -0,0 +1,55 @@
+#*************************************************************************
+#
+#   OpenOffice.org - a multi-platform office productivity suite
+#
+#   $RCSfile: makefile.mk,v $
+#
+#   $Revision: 1.3 $
+#
+#   last change: $Author: rt $ $Date: 2005/09/08 02:44:02 $
+#
+#   The Contents of this file are made available subject to
+#   the terms of GNU Lesser General Public License Version 2.1.
+#
+#
+#     GNU Lesser General Public License Version 2.1
+#     =============================================
+#     Copyright 2005 by Sun Microsystems, Inc.
+#     901 San Antonio Road, Palo Alto, CA 94303, USA
+#
+#     This library is free software; you can redistribute it and/or
+#     modify it under the terms of the GNU Lesser General Public
+#     License version 2.1, as published by the Free Software Foundation.
+#
+#     This library is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#     Lesser General Public License for more details.
+#
+#     You should have received a copy of the GNU Lesser General Public
+#     License along with this library; if not, write to the Free Software
+#     Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+#     MA  02111-1307  USA
+#
+#*************************************************************************
+
+PRJ=..$/..
+PRJINC=..$/..$/inc
+PRJNAME=comphelper
+TARGET=unitbootstrap
+
+ENABLE_EXCEPTIONS=TRUE
+
+# --- Settings ----------------------------------
+
+.INCLUDE : settings.mk
+.INCLUDE : $(PRJ)$/util$/makefile.pmk
+
+# --- Files -------------------------------------
+
+SLOFILES=$(SLO)$/unitbootstrap.obj
+
+# --- Targets ----------------------------------
+
+.INCLUDE : target.mk
+
--- /dev/null	2007-09-21 22:50:58.000000000 +0100
+++ comphelper/source/unit/Setup.xcu	2008-02-20 16:57:42.000000000 +0000
@@ -0,0 +1,376 @@
+<?xml version="1.0"?>
+<oor:component-data xmlns:oor="http://openoffice.org/2001/registry" xmlns:install="http://openoffice.org/2004/installation" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" oor:name="Setup" oor:package="org.openoffice">
+  <node oor:name="Product">
+    <prop oor:name="ooName">
+      <value>${PRODUCTNAME}</value>
+    </prop>
+    <prop oor:name="ooSetupVersion">
+      <value>${PRODUCTVERSION}</value>
+    </prop>
+    <prop oor:name="ooSetupVersionAboutBox">
+      <value>${ABOUTBOXPRODUCTVERSION}</value>
+    </prop>
+    <prop oor:name="ooSetupExtension">
+      <value>${PRODUCTEXTENSION}</value>
+    </prop>
+    <prop oor:name="ooXMLFileFormatVersion">
+      <value>${FILEFORMATVERSION}</value>
+    </prop>
+    <prop oor:name="ooXMLFileFormatName">
+      <value>${FILEFORMATNAME}</value>
+    </prop>
+    <prop oor:name="ooOpenSourceContext">
+      <value>${OPENSOURCE}</value>
+    </prop>
+  </node>
+  <node oor:name="Office">
+    <prop oor:name="ooSetupInstCompleted">
+      <value>false</value>
+    </prop>
+    <prop oor:name="ooSetupShowIntro">
+      <value>true</value>
+    </prop>
+    <node oor:name="Factories">
+      <node oor:name="com.sun.star.text.WebDocument" oor:op="replace">
+        <prop oor:name="ooSetupFactoryDocumentService" oor:finalized="true">
+          <value>com.sun.star.text.WebDocument</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryCommandConfigRef">
+          <value>WriterCommands</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryActualFilter" oor:finalized="true">
+          <value>HTML</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryActualTemplateFilter" oor:finalized="true"/>
+        <prop oor:name="ooSetupFactoryDefaultFilter">
+          <value>HTML</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryEmptyDocumentURL" oor:finalized="true">
+          <value>private:factory/swriter/web</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryWindowAttributes">
+          <value/>
+        </prop>
+        <prop oor:name="ooSetupFactoryIcon">
+          <value>12</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryTemplateFile">
+          <value/>
+        </prop>
+        <prop oor:name="ooSetupFactoryShortName">
+          <value>sweb</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryUIName">
+          <value>Writer/Web</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryWindowStateConfigRef">
+          <value>WriterWebWindowState</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryCmdCategoryConfigRef">
+          <value>GenericCategories</value>
+        </prop>
+      </node>
+      <node oor:name="com.sun.star.frame.Bibliography" oor:op="replace">
+        <prop oor:name="ooSetupFactoryDocumentService" oor:finalized="true"/>
+        <prop oor:name="ooSetupFactoryActualFilter" oor:finalized="true"/>
+        <prop oor:name="ooSetupFactoryActualTemplateFilter" oor:finalized="true"/>
+        <prop oor:name="ooSetupFactoryDefaultFilter" oor:finalized="true"/>
+        <prop oor:name="ooSetupFactoryEmptyDocumentURL" oor:finalized="true"/>
+        <prop oor:name="ooSetupFactoryCommandConfigRef">
+          <value>BibliographyCommands</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryShortName">
+          <value>sbibliography</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryWindowStateConfigRef">
+          <value>BibliographyWindowState</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryCmdCategoryConfigRef">
+          <value>GenericCategories</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryUIName">
+          <value>Bibliography</value>
+        </prop>
+      </node>
+      <node oor:name="com.sun.star.script.BasicIDE" oor:op="replace">
+        <prop oor:name="ooSetupFactoryDocumentService" oor:finalized="true"/>
+        <prop oor:name="ooSetupFactoryActualFilter" oor:finalized="true"/>
+        <prop oor:name="ooSetupFactoryActualTemplateFilter" oor:finalized="true"/>
+        <prop oor:name="ooSetupFactoryDefaultFilter" oor:finalized="true"/>
+        <prop oor:name="ooSetupFactoryEmptyDocumentURL" oor:finalized="true"/>
+        <prop oor:name="ooSetupFactoryCommandConfigRef">
+          <value>BasicIDECommands</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryShortName">
+          <value>BasicIDE</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryWindowStateConfigRef">
+          <value>BasicIDEWindowState</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryCmdCategoryConfigRef">
+          <value>GenericCategories</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryUIName">
+          <value>Basic</value>
+        </prop>
+      </node>
+      <node oor:name="com.sun.star.frame.StartModule" oor:op="replace">
+        <prop oor:name="ooSetupFactoryDocumentService" oor:finalized="true"/>
+        <prop oor:name="ooSetupFactoryActualFilter" oor:finalized="true"/>
+        <prop oor:name="ooSetupFactoryActualTemplateFilter" oor:finalized="true"/>
+        <prop oor:name="ooSetupFactoryDefaultFilter" oor:finalized="true"/>
+        <prop oor:name="ooSetupFactoryEmptyDocumentURL" oor:finalized="true"/>
+        <prop oor:name="ooSetupFactoryCommandConfigRef">
+          <value>StartModuleCommands</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryShortName">
+          <value>StartModule</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryWindowStateConfigRef">
+          <value>StartModuleWindowState</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryCmdCategoryConfigRef">
+          <value>GenericCategories</value>
+        </prop>
+      </node>
+      <node oor:name="com.sun.star.chart2.ChartDocument" oor:op="replace">
+        <prop oor:name="ooSetupFactoryDocumentService" oor:finalized="true">
+          <value>com.sun.star.chart2.ChartDocument</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryCommandConfigRef">
+          <value>ChartCommands</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryActualFilter" oor:finalized="true">
+          <value>chart8</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryActualTemplateFilter" oor:finalized="true"/>
+        <prop oor:name="ooSetupFactoryDefaultFilter">
+          <value>chart8</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryEmptyDocumentURL" oor:finalized="true">
+          <value>private:factory/schart</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryWindowAttributes">
+          <value/>
+        </prop>
+        <prop oor:name="ooSetupFactoryIcon">
+          <value>13</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryTemplateFile">
+          <value/>
+        </prop>
+        <prop oor:name="ooSetupFactoryShortName">
+          <value>schart</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryUIName">
+          <value>Chart</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryWindowStateConfigRef">
+          <value>ChartWindowState</value>
+        </prop>
+        <prop oor:name="ooSetupFactoryCmdCategoryConfigRef">
+          <value>GenericCategories</value>
+        </prop>
+      </node>
+    </node>
+  </node>
+  <node oor:name="L10N"/>
+  <node oor:name="Configuration">
+    <prop oor:name="TransferUserSettingsOnce" oor:type="xs:boolean">
+      <value>true</value>
+    </prop>
+  </node>
+  <node oor:name="Migration">
+    <prop oor:name="SupportedVersions">
+      <value oor:separator=",">
+                        StarOffice 7, OpenOffice.org 1.1, StarSuite 7
+		    </value>
+    </prop>
+    <node oor:name="MigrationSteps">
+      <node oor:name="Datasources" oor:op="replace">
+        <prop oor:name="MigrationService">
+          <value>com.sun.star.comp.sdb.DataSourceMigration</value>
+        </prop>
+        <prop oor:name="ServiceConfigComponents">
+          <value>org.openoffice.Office.DataAccess</value>
+        </prop>
+        <prop oor:name="IncludedFiles">
+          <value>.*/database/biblio/biblio\.dbf</value>
+        </prop>
+        <prop oor:name="ExcludedFiles"/>
+        <prop oor:name="IncludedNodes">
+          <value>
+                        org.openoffice.Office.DataAccess
+                    </value>
+        </prop>
+        <prop oor:name="ExcludedNodes">
+          <value>
+                        org.openoffice.Office.DataAccess/Bibliography
+                        org.openoffice.Office.DataAccess/ConnectionPool
+                        org.openoffice.Office.DataAccess/DataSources
+                        org.openoffice.Office.DataAccess/DriverManager
+                    </value>
+        </prop>
+      </node>
+      <node oor:name="Deployment" oor:op="replace">
+        <prop oor:name="MigrationService">
+          <value>com.sun.star.comp.deployment.migration.Migration_2_0</value>
+        </prop>
+      </node>
+      <node oor:name="Inet" oor:op="replace">
+        <prop oor:name="IncludedNodes">
+          <value>org.openoffice.Inet
+					</value>
+        </prop>
+      </node>
+      <node oor:name="Java" oor:op="replace">
+        <prop oor:name="MigrationService">
+          <value>com.sun.star.migration.Java</value>
+        </prop>
+        <prop oor:name="ServiceConfigComponents">
+          <value>org.openoffice.Office.Java</value>
+        </prop>
+      </node>
+      <node oor:name="Basic" oor:op="replace">
+        <prop oor:name="MigrationService">
+          <value>com.sun.star.migration.Basic</value>
+        </prop>
+      </node>
+      <node oor:name="Autocorrection" oor:op="replace">
+        <prop oor:name="MigrationService">
+          <value>com.sun.star.migration.Autocorrection</value>
+        </prop>
+      </node>
+      <node oor:name="UserProfile" oor:op="replace">
+        <prop oor:name="IncludedNodes">
+          <value>org.openoffice.UserProfile</value>
+        </prop>
+      </node>
+      <node oor:name="Common" oor:op="replace">
+        <prop oor:name="IncludedFiles">
+          <value>
+                        .*/autotext/.*
+                        .*/config/.*\.so[bcdegh]
+                        .*/gallery/.*
+                        .*/template/.*
+                    </value>
+        </prop>
+        <prop oor:name="IncludedNodes">
+          <value>
+                        org.openoffice.Office.Common/Accessibility
+                        org.openoffice.Office.Common/Accessibility/AutoDetectSystemHC
+                        org.openoffice.Office.Common/AddXMLToStrorage
+                        org.openoffice.Office.Common/AsianLayout
+                        org.openoffice.Office.Common/AutoCorrect
+                        org.openoffice.Office.Common/Cache
+                        org.openoffice.Office.Common/ExternalMailer/Program
+                        org.openoffice.Office.Common/Filter
+                        org.openoffice.Office.Common/Font
+                        org.openoffice.Office.Common/Gallery
+                        org.openoffice.Office.Common/Help
+                        org.openoffice.Office.Common/History
+                        org.openoffice.Office.Common/I18N
+                        org.openoffice.Office.Common/InternalMSExport
+                        org.openoffice.Office.Common/Load
+                        org.openoffice.Office.Common/Misc/Form/ControlPilotsEnabled
+                        org.openoffice.Office.Common/Misc/PluginsEnabled
+                        org.openoffice.Office.Common/Misc/SymbolSet
+                        org.openoffice.Office.Common/Misc/UseSystemFileDialog
+                        org.openoffice.Office.Common/Misc/SymbolStyle
+                        org.openoffice.Office.Common/OfficeObjects
+                        org.openoffice.Office.Common/Passwords
+                        org.openoffice.Office.Common/Print/PrintingModifiesDocument
+                        org.openoffice.Office.Common/Print/Warning
+                        org.openoffice.Office.Common/PrintVectorize
+                        org.openoffice.Office.Common/Save
+                        org.openoffice.Office.Common/SearchOptions
+                        org.openoffice.Office.Common/SearchOptionsJapanese
+                        org.openoffice.Office.Common/Undo
+                        org.openoffice.Office.Common/View/Dialog/Dialog/MiddleMouseButton
+                        org.openoffice.Office.Common/View/Dialog/MousePositioning
+                        org.openoffice.Office.Common/View/Localization
+                        org.openoffice.Office.Common/View/Menu
+                        org.openoffice.Office.Common/_3D_Engine
+                    </value>
+        </prop>
+        <prop oor:name="ExcludedNodes">
+          <value>
+                        org.openoffice.Office.Common/Filter/PDF
+                        org.openoffice.Office.Common/_3D_Engine/OpenGL
+                    </value>
+        </prop>
+      </node>
+      <node oor:name="Calc" oor:op="replace">
+        <prop oor:name="IncludedNodes">
+          <value>org.openoffice.Office.Calc</value>
+        </prop>
+      </node>
+      <node oor:name="Chart" oor:op="replace">
+        <prop oor:name="IncludedNodes">
+          <value>org.openoffice.Office.Chart/DefaultColor</value>
+        </prop>
+      </node>
+      <node oor:name="Draw" oor:op="replace">
+        <prop oor:name="IncludedNodes">
+          <value>org.openoffice.Office.Draw</value>
+        </prop>
+      </node>
+      <node oor:name="Impress" oor:op="replace">
+        <prop oor:name="IncludedNodes">
+          <value>org.openoffice.Office.Impress</value>
+        </prop>
+      </node>
+      <node oor:name="Labels" oor:op="replace">
+        <prop oor:name="IncludedNodes">
+          <value>org.openoffice.Office.Labels</value>
+        </prop>
+      </node>
+      <node oor:name="Linguistic" oor:op="replace">
+        <prop oor:name="IncludedFiles">
+          <value>.*/wordbook/.*</value>
+        </prop>
+        <prop oor:name="IncludedNodes">
+          <value>org.openoffice.Office.Linguistic</value>
+        </prop>
+        <prop oor:name="ExcludedNodes">
+          <value>
+                        org.openoffice.Office.Linguistic/ServiceManager
+                    </value>
+        </prop>
+      </node>
+      <node oor:name="Math" oor:op="replace">
+        <prop oor:name="IncludedNodes">
+          <value>org.openoffice.Office.Math</value>
+        </prop>
+      </node>
+      <node oor:name="Security" oor:op="replace">
+        <prop oor:name="IncludedNodes">
+          <value>org.openoffice.Office.Security</value>
+        </prop>
+      </node>
+      <node oor:name="Setup" oor:op="replace">
+        <prop oor:name="IncludedNodes">
+          <value>org.openoffice.Office.Setup/L10N</value>
+        </prop>
+      </node>
+      <node oor:name="UI" oor:op="replace">
+        <prop oor:name="IncludedNodes">
+          <value>org.openoffice.Office.UI/ColorScheme</value>
+        </prop>
+      </node>
+      <node oor:name="Writer" oor:op="replace">
+        <prop oor:name="IncludedNodes">
+          <value>
+                        org.openoffice.Office.Writer
+					    org.openoffice.Office.WriterWeb
+                    </value>
+        </prop>
+        <prop oor:name="ExcludedNodes">
+          <value>
+                        org.openoffice.Office.Writer/Wizard
+                    </value>
+        </prop>
+      </node>
+    </node>
+  </node>
+</oor:component-data>
--- /dev/null	2007-09-21 22:50:58.000000000 +0100
+++ comphelper/source/unit/configmgrrc	2008-02-20 11:02:31.000000000 +0000
@@ -0,0 +1,22 @@
+[Bootstrap]
+CFG_ServerType=uno
+BootstrapFile=$ORIGIN/bootstraprc
+VersionFile=$ORIGIN/versionrc
+UnoFile=$ORIGIN/unorc
+BaseInstallation=${$BootstrapFile:BaseInstallation}
+UserInstallation=${$BootstrapFile:UserInstallation}
+CFG_SchemaVersion=${$VersionFile:buildid}
+CFG_SchemaDataUrl=$SOLVER/xml/registry/schema
+CFG_DefaultLayerUrls=$SOLVER/xml/registry
+CFG_CacheDisposeDelay=900
+CFG_CacheDisposeInterval=60
+CFG_CacheWriteInterval=2
+CFG_UserLayerUrl=$UserInstallation
+CFG_CacheUrl=$UserInstallation/cache
+CFG_DefaultLayerUrl=$SOLVER/xml/registry
+CFG_BackendService=com.sun.star.comp.configuration.backend.MultiStratumBackend
+CFG_SchemaSupplier=com.sun.star.comp.configuration.backend.LocalSchemaSupplier
+CFG_LocalDataSvc=com.sun.star.comp.configuration.backend.LocalStratum
+CFG_LocalModuleDataSvc=com.sun.star.comp.configuration.backend.LocalMultiStratum
+# CFG_Strata=${CFG_LocalDataSvc}:$CFG_DefaultLayerUrl;?com.sun.star.comp.configuration.backend.SystemIntegration:;?com.sun.star.comp.configuration.backend.PolicyLayerBackend:;${CFG_LocalDataSvc}:$CFG_UserLayerUrl
+CFG_Strata=${CFG_LocalDataSvc}:$CFG_DefaultLayerUrl;${CFG_LocalDataSvc}:$CFG_UserLayerUrl
--- /dev/null	2007-09-21 22:50:58.000000000 +0100
+++ comphelper/source/unit/README	2008-02-20 16:56:45.000000000 +0000
@@ -0,0 +1 @@
+This includes pieces required to bootstrap a unit test from a build tree, cf. prj/d.lst
